File: zariz/ios/Zariz/Data/Models/Order.swift
import Foundation
import SwiftData

@Model
final class OrderEntity {
    @Attribute(.unique) var id: Int
    var status: String
    var pickupAddress: String
    var deliveryAddress: String
    var recipientFirstName: String
    var recipientLastName: String
    var recipientPhone: String
    var street: String
    var buildingNumber: String
    var floor: String?
    var apartment: String?
    var boxesCount: Int
    var boxesMultiplier: Int
    var priceTotal: Double

    init(
        id: Int,
        status: String,
        pickupAddress: String,
        deliveryAddress: String,
        recipientFirstName: String = "",
        recipientLastName: String = "",
        recipientPhone: String = "",
        street: String = "",
        buildingNumber: String = "",
        floor: String? = nil,
        apartment: String? = nil,
        boxesCount: Int = 0,
        boxesMultiplier: Int = 1,
        priceTotal: Double = 0
    ) {
        self.id = id
        self.status = status
        self.pickupAddress = pickupAddress
        self.deliveryAddress = deliveryAddress
        self.recipientFirstName = recipientFirstName
        self.recipientLastName = recipientLastName
        self.recipientPhone = recipientPhone
        self.street = street
        self.buildingNumber = buildingNumber
        if let floor, !floor.trimmingCharacters(in: .whitespaces).isEmpty {
            self.floor = floor
        } else {
            self.floor = nil
        }
        if let apartment, !apartment.trimmingCharacters(in: .whitespaces).isEmpty {
            self.apartment = apartment
        } else {
            self.apartment = nil
        }
        self.boxesCount = boxesCount
        self.boxesMultiplier = boxesMultiplier
        self.priceTotal = priceTotal
    }

    var recipientFullName: String {
        [recipientFirstName, recipientLastName]
            .map { $0.trimmingCharacters(in: .whitespaces) }
            .filter { !$0.isEmpty }
            .joined(separator: " ")
    }
}

File: zariz/backend/app/db/models/order.py
from sqlalchemy import Integer, String, ForeignKey, DateTime, func
from sqlalchemy.orm import Mapped, mapped_column
from ..base import Base


class Order(Base):
    __tablename__ = "orders"
    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    store_id: Mapped[int] = mapped_column(ForeignKey("stores.id"))
    courier_id: Mapped[int | None] = mapped_column(ForeignKey("users.id"), nullable=True)
    status: Mapped[str] = mapped_column(String(32), index=True)  # new/assigned/claimed/picked_up/delivered/canceled
    pickup_address: Mapped[str] = mapped_column(String(255))
    delivery_address: Mapped[str] = mapped_column(String(255))
    recipient_first_name: Mapped[str] = mapped_column(String(120))
    recipient_last_name: Mapped[str] = mapped_column(String(120))
    phone: Mapped[str] = mapped_column(String(40))
    street: Mapped[str] = mapped_column(String(180))
    building_no: Mapped[str] = mapped_column(String(30))
    floor: Mapped[str] = mapped_column(String(30), default="")
    apartment: Mapped[str] = mapped_column(String(30), default="")
    boxes_count: Mapped[int] = mapped_column(Integer, default=0)
    boxes_multiplier: Mapped[int] = mapped_column(Integer, default=1)
    price_total: Mapped[int] = mapped_column(Integer, default=0)
    created_at: Mapped[DateTime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[DateTime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
